src/libinjector/linux/linux_utils.h

```diff
@@ -0,0 +1,67 @@
+#ifndef LINUX_UTILS_H
+#define LINUX_UTILS_H
+
+#include <errno.h>
+#include <glib.h>
+#include <inttypes.h>
+#include <json-c/json.h>
+#include <libinjector/libinjector.h>
+#include <libvmi/libvmi.h>
+#include <libvmi/x86.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+
+#include "libdrakvuf/libdrakvuf.h"
+#include "libinjector/private.h"
+
+typedef enum
+{
+    INJECT_RESULT_SUCCESS,
+    INJECT_RESULT_TIMEOUT,
+    INJECT_RESULT_CRASH,
+    INJECT_RESULT_PREMATURE,
```

Comment:
Not sure what is INJECT_RESULT_PREMATURE

Reply:
it was basically called when we use `SIGINT` to terminate the injector before it is successful afair, though i haven't added any checks for that yet

Reply:
If used, then for now let's remove it. If you want to use it, then rename it to something indicating that it's SIGINT.

Reply:
i will remove it for now then